lxc/storage_volume.go

```diff
@@ -373,10 +399,18 @@ func (c *cmdStorageVolumeCopy) Run(cmd *cobra.Command, args []string) error {
 	dstServer := dstResource.server
 	dstPath := dstResource.name
 
-	// Get source pool and volume name
-	srcVolName, srcVolPool := c.storageVolume.parseVolumeWithPool(srcPath)
-	if srcVolPool == "" {
-		return fmt.Errorf(i18n.G("No storage pool for source volume specified"))
+	// We can always set the destination target if the destination server is clustered (for local storage volumes this
+	// places the volume on the target member, for remote volumes this does nothing).
+	if c.storageVolume.flagDestinationTarget != "" {
+		if !dstServer.IsClustered() {
+			return fmt.Errorf(i18n.G("Cannot set --destination-target when destination server is not clustered"))
+		}
+
+		dstServer = dstServer.UseTarget(c.storageVolume.flagDestinationTarget)
+	} else if srcResource.remote == dstResource.remote {
+		// If moving within the same remote and no --destination-target flag is given, set the target to be the same as
+		// the source volume location.
+		dstServer = dstServer.UseTarget(srcVol.Location)
```

Comment:
@tomponline I've added this to get the tests passing again. It makes sense for a clustered setup, but in the standalone case, this sets the `target` to "none" which is a little strange but still necessary (otherwise [this predicate in the client](volume.Location == r.clusterTarget) returns false and we don't enter the if block).

Reply:
I wonder if, instead of
```go
volume.Location == r.clusterTarget
```
the client should have something like
```go
volume.Location == r.clusterTarget || (volume.Location == "none" && r.clusterTarget == "")
```

Reply:
yeah I think that makes sense and would be clearer and safer.

Reply:
Ok I've made that change. I *think* we should be good to go - so many scenarios to consider when moving storage volumes, this definitely wasn't as simple as I originally thought :laughing:.